src/hotspot/share/jvmci/jvmciEnv.cpp

```diff
@@ -316,20 +316,29 @@ class ExceptionTranslation: public StackObj {
     int buffer_size = 2048;
     while (true) {
       ResourceMark rm;
-      jlong buffer = (jlong) NEW_RESOURCE_ARRAY_IN_THREAD(THREAD, jbyte, buffer_size);
-      int res = encode(THREAD, runtimeKlass, buffer, buffer_size);
-      if ((_from_env != nullptr && _from_env->has_pending_exception()) || HAS_PENDING_EXCEPTION) {
-        JVMCIRuntime::fatal_exception(_from_env, "HotSpotJVMCIRuntime.encodeThrowable should not throw an exception");
+      jlong buffer = (jlong) NEW_RESOURCE_ARRAY_IN_THREAD_RETURN_NULL(THREAD, jbyte, buffer_size);
+      if (buffer == 0L) {
+        decode(THREAD, runtimeKlass, 0L);
```

Comment:
Can we add an argument so that each of these call sites reports a unique message?  Should we get the class name from the pending exception and include that as well?  I think we should include enough breadcrumbs for future failures in the this path that we might have a better guess what's happening.

Reply:
I think the case we care about most is an `OutOfMemoryError` occurring in the HotSpot heap so I've pushed a change that calls this out. I've also distinguished the case where the native buffer for the encoding cannot be allocated (although if that happens, the VM is in real trouble).

Reply:
I was thinking we could get the jthrowable and somehow find the name of the exception from the jclass but I guess JNI doesn't let you do that.  Should we use `_from_env->describe_pending_exception()`?

Reply:
Good idea but unfortunately the [JNI `ExceptionDescribe` function](https://docs.oracle.com/en/java/javase/19/docs/specs/jni/functions.html#exceptiondescribe) writes to the console which is something we don't want in this context.

Reply:
If we were still treating it as fatal it would be fine but I agree it's not ideal in the new code.